### DISCUSSION TITLE
using existing monads

### DISCUSSION THREAD

**Discussion by drlkf** [haskell-effectful/effectful#163] [Q&A]
How would you recommend using existing monads such as the `brick` [EventM](https://hackage.haskell.org/package/brick-1.9/docs/Brick-Types.html#t:EventM) monad? From what I understand, the recommended way is to [create boilerplate](https://hackage.haskell.org/package/effectful-core-2.2.2.2/docs/Effectful-Dispatch-Dynamic.html#g:4) for each monad instance. Is this correct? If so, are there some other packages implementing common use-cases such as RWS-type monads and such?

**Comment by ocharles** [comment]
Can you share a bit more about what it is you want to do?

**Reply by drlkf** [reply]
I have an existing application that currently uses `polysemy` to handle effects. I'm building an interactive terminal frontend using some existing functions that would have e.g. `Members '[Final IO, Logger, Error MyError]` and I would like to:
- build interpreters to handle my effects from within `EventM` handlers (e.g. `Logger` would add a log message to the logs pane
- wrap all the events in some equivalent to `Member (Embed EventM)` and be able to use `EventM` native capabilities such as `State` and `Zoom`, while also adding my own effects such as `Logger` using the aforementioned interpreters in [`appHandleEvent`](https://hackage.haskell.org/package/brick-1.9/docs/Brick-Main.html#t:App)
